Ota firmware updates

kYc0o opened the pull request.

Commit:
rpl: add DIS Solicited Information option

kaspar030 committed:
make: initial multislot support

kaspar030 committed:
riotboot: introduce riot-based minimal bootloader

kaspar030 committed:
bootloader: add initial minimal bootloader

kaspar030 committed:
cpu/cortexm_common: initial multislot support

kaspar030 committed:
sys/firmware: initial import of firmware management functions

kaspar030 committed:
sys/hashes/sha256: add missing unistd.h include

kaspar030 committed:
dist/tools/firmware: initial import of firmware metadata tools

kYc0o committed:
tests: add bootloader test

kYc0o committed:
dist/tools/openocd: add flash-bin option

kaspar030 commented on sys/ota_update/ota_update.c:
```diff
@@ -0,0 +1,494 @@
+/*
```
please split into tftp-server and coap+tftp-client

kYc0o replied:
Done.

kaspar030 commented on sys/include/ota_update.h:
```diff
@@ -0,0 +1,120 @@
+#define OTA_UPDATE_STOP           (0x1000)
+
+/**
+ * @brief Peiodic request time for update
```
periodic

kaspar030 commented on boards/iotlab-common/Makefile.features:
```diff
@@ -11,5 +11,9 @@ FEATURES_PROVIDED += periph_uart
 # Various other features (if any)
 FEATURES_PROVIDED += cpp
 
+ifdef APP_ID
```
please remove ifdef

kaspar030 committed:
cpu/cortexm_common: set MSP instead of PSP in cpu_jump_to_image()

kaspar030 commented:
please rebase to master!

kYc0o commented:
Done.

kaspar030 committed:
edbg: add FLASH_ADDR as offset, don't erase before flash

kaspar030 committed:
cpu/cortexm: change rom offset logic in cortexm.ld

kaspar030 committed:
boards/samr21-xpro: add riotboot configuration

kYc0o committed:
cpu/sam0_common: add metadata size for binary updates

kaspar030 committed:
sys/ota: introduce CoAP OTA module

kaspar030 committed:
examples/ota: initial commit of OTA example

bergzand committed:
nanocoap: Add server-side block2 support

bergzand committed:
examples/nanocoap: add blockwise block2 example

bergzand committed:
firmware: don't depend on sizeof(firmware_metadata_t)

bergzand committed:
Split update into firmware_simple and firmware_flashwrite

bergzand committed:
firmware: remove metadata_type from struct

bergzand committed:
nanocoap_sock: split request function from get

The nanocoap_get function is refactored to split of the request part
into a separate function for reuse by other modules. Support for
retransmissions when the received frame is malformed is dropped as it
was broken anyway.

bergzand committed:
suit: Initial include of SUIT manifest parser

bergzand committed:
Add conditional support

bergzand committed:
ota_suit: Initial work

bergzand committed:
getters for the uuids

bergzand committed:
Add simple url extraction

bergzand committed:
url parsing

cgundogan committed:
Merge pull request #6918 from BytesGalore/rpl_add_solicited_information_option

rpl: add solicited information option to DIS

bergzand committed:
Update

cladmi committed:
fixup! Split update into firmware_simple and firmware_flashwrite
